Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionDataRecord.h

```diff
@@ -30,8 +30,29 @@
 
 #import <WebKit/_WKWebExtensionDataType.h>
 
+#define HAVE_UPDATED_WEB_EXTENSION_DATA_RECORD_ERROR_PROPERTIES 1
+
 NS_ASSUME_NONNULL_BEGIN
 
+/*! @abstract Indicates a `_WKWebExtensionDataRecord` error. */
+WK_EXTERN NSErrorDomain const _WKWebExtensionDataRecordErrorDomain WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
+
+/*!
+ @abstract Constants used by NSError to indicate errors in the `_WKWebExtensionDataRecord` domain.
+ @constant WKWebExtensionDataRecordErrorUnknown  Indicates that an unknown error occurred.
+
+ @constant WKWebExtensionDataRecordErrorLocalStorageFailed  Indicates a failure occurred when either deleting or calculating local storage.
+ @constant WKWebExtensionDataRecordErrorSessionStorageFailed  Indicates a failure occurred when either deleting or calculating session storage.
+ @constant WKWebExtensionDataRecordErrorSyncStorageFailed  Indicates a failure occurred when either deleting or calculating sync storage.
+
+ */
+typedef NS_ERROR_ENUM(_WKWebExtensionDataRecordErrorDomain, _WKWebExtensionDataRecordError) {
+    _WKWebExtensionDataRecordErrorUnknown,
+    _WKWebExtensionDataRecordErrorLocalStorageFailed,
+    _WKWebExtensionDataRecordErrorSessionStorageFailed,
+    _WKWebExtensionDataRecordErrorSyncStorageFailed,
```

Comment:
I think we want to have either an `errors` array getter, that returns all applicable errors (my preference), or you only need one error code, since you have individual error properties, and `_WKWebExtensionDataRecordErrorLocalStorageFailed` will always be the case for `localStorageError`, etc.

Again, I think if we have a `errors` property, then these should stay as-is.